src/nodes/NodeManager.ts

```diff
@@ -36,37 +36,67 @@ class NodeManager {
   protected keyManager: KeyManager;
   protected nodeConnectionManager: NodeConnectionManager;
   protected nodeGraph: NodeGraph;
-  protected queue: Queue;
-  // Refresh bucket timer
-  protected refreshBucketDeadlineMap: Map<NodeBucketIndex, number> = new Map();
-  protected refreshBucketTimer: NodeJS.Timer;
-  protected refreshBucketNext: NodeBucketIndex;
-  public readonly refreshBucketTimerDefault;
-  protected refreshBucketQueue: Set<NodeBucketIndex> = new Set();
-  protected refreshBucketQueueRunning: boolean = false;
-  protected refreshBucketQueueRunner: Promise<void>;
-  protected refreshBucketQueuePlug_: PromiseDeconstructed<void> = promise();
-  protected refreshBucketQueueDrained_: PromiseDeconstructed<void> = promise();
-  protected refreshBucketQueuePause_: PromiseDeconstructed<void> = promise();
-  protected refreshBucketQueueAbortController: AbortController;
+  protected taskManager: TaskManager;
+  protected refreshBucketDelay: number;
+  protected refreshBucketDelaySpread: number;
+
+  private refreshBucketHandler: TaskHandler = async (
+    context,
+    taskInfo,
+    bucketIndex,
+  ) => {
+    await this.refreshBucket(bucketIndex, { signal: context.signal });
+    // When completed reschedule the task
+    const spread =
+      (Math.random() - 0.5) *
+      this.refreshBucketDelay *
+      this.refreshBucketDelaySpread;
+    await this.taskManager.scheduleTask({
+      delay: this.refreshBucketDelay + spread,
+      handlerId: this.refreshBucketHandlerId,
+      lazy: true,
+      parameters: [bucketIndex],
+      path: ['refreshBucket', `${bucketIndex}`],
```

Comment:
This should become a class property. It should also have the class name as the prefix. Similar to the task handler name as well. That means using `this.constructor.name`.

Reply:
This has been addressed for all handlers.